src/components/forms/CharacterCount/CharacterCount.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useState } from 'react'
+import classnames from 'classnames'
+
+import { TextInput, TextInputRef } from '../TextInput/TextInput'
+import { Textarea, TextareaRef } from '../Textarea/Textarea'
+import { FormGroup } from '../FormGroup/FormGroup'
+
+interface CharacterCountProps {
+  id: string
+  name: string
+  maxLength: number
+  className?: string
+  label?: React.ReactNode
+  isTextArea?: boolean
+  textRef?: // currently getting an error when adding this as ref
+  | string
+    | ((instance: HTMLInputElement | null) => void)
+    | ((instance: HTMLTextAreaElement | null) => void)
+    | React.RefObject<HTMLInputElement>
+    | React.RefObject<HTMLTextAreaElement>
+    | null
+    | undefined
+}
+
+export const CharacterCount = (
+  props: CharacterCountProps &
+    React.InputHTMLAttributes<HTMLInputElement> &
+    React.TextareaHTMLAttributes<HTMLTextAreaElement>
+): React.ReactElement => {
+  const {
+    id,
+    name,
+    className,
+    label,
+    maxLength,
+    defaultValue,
+    isTextArea,
+    textRef,
+    type, // Conflicts with type attribute already set on TextInput component
```

Comment:
Another way to deal with this is just in order of props.
```
<TextInput
        id={id}
        data-testid="characterCountInput"
        name={name}
        className={classes}
        defaultValue={defaultValue}
        onChange={handleChange}
        onBlur={handleBlur}
        aria-describedby={messageId}
        inputRef={ref}
        {...inputProps}
        type="text"
      />
```

We could also write a test to make sure when you pass in a different type it doesn't apply (and document the expected behavior that way rather than through a comment).

Reply:
the value of `type` should be whatever the user passed in via `inputProps`.. why does this need to be explicitly passed in here?